ludwig/profiling/dataset_profile.py

```diff
@@ -0,0 +1,74 @@
+import time
+from typing import Dict, Tuple, Union
+
+import whylogs as why
+from whylogs.core.proto import ColumnMessage
+from whylogs.core.view.column_profile_view import ColumnProfileView
+from whylogs.core.view.dataset_profile_view import DatasetProfileView
+
+from ludwig.profiling import dataset_profile_pb2
+from ludwig.profiling.types import ColumnProfileSummary
+from ludwig.utils.data_utils import load_dataset
+from ludwig.utils.types import DataFrame
+
+# Absolute cap on the data that is used for profiling.
+PROFILING_CAP = 100000
+
+
+def get_dataset_profile_view(dataset: Union[str, DataFrame]) -> Tuple[DatasetProfileView, int]:
+    """Returns whylogs dataset profile view."""
+    dataframe = load_dataset(dataset)
+    size_bytes = sum(dataframe.memory_usage(deep=True))
+
+    # Manual cap, also takes care of converting dask to pandas.
+    dataframe = dataframe.head(PROFILING_CAP)
```

Comment:
Do we need to have a cap, and if so should this be an optional argument?

If we were to cap, instead of taking `.head` here should we use `.sample(PROFILING_CAP)` if the `shape[0]> PROFILING_CAP`  so that we a better distribution when the data is sorted like in the case of timeseries data?

Reply:
`sample()` is a great idea, though using `.head()` here to be consistent with what we're doing in pbase, and avoids a few issues detailed below:

1. If you do `.sample(n)`, `n` needs to be strictly less than the total number of rows in the dataframe, else we run into this error.

> `ValueError: Cannot take a larger sample than population when 'replace=False'`

`replace=False` seems not ideal.

2. `sample()` with a dask dataframe seems to require a ratio instead of a pure number, which could be tricky to get right without checking the `len()` of the overall dataframe, which I believe is an expensive call that we want to try to avoid (cc: @dantreiman @tgaddair). The error:

> `results = ValueError("sample does not support the number of sampled items parameter, 'n'. Please use the 'frac' parameter instead.")`

Making `cap` an optional parameter SGTM though.